File: Evaluacion/algoritmos_extraccion/run_rvest.R
library(rvest)
library(rjson)
library(stringr)
library(utf8)

rvest_json <- function(output_text) {
    ###almacena el texto obtenido de rvest en un fichero json###

    write(toJSON(output_text, indent = 4), "./archivos_salida/rvest.json")
}

run_rvest <- function() {
    ###extraccion de texto empleando rvest###

    output <- c()
    files <- list.files(path = "./archivos_html", pattern = ".html") #corpus de prueba
    index <- 1
    for (key in fromJSON(file = "./documento_base.json")) {
        #extraccion y almacenamiento de texto
        html_to_string <- read_html(key$url, encoding = "UTF-8") %>% html_nodes(xpath = '//body') %>% html_text() %>% utf8_encode()
        lista_texto <- list('xxxxtest' = list("texto" = gsub("\\r|\\t|\\n|\  ", "", unlist(html_to_string))))

        names(lista_texto) <- str_remove(files[index], ".html")
        output <- c(output, lista_texto)
        index <- index + 1
    }
    rvest_json(output)
}

#Probador
run_rvest()

File: Evaluacion/algoritmos_extraccion/run_Rvest.py
import json

import psutil
import time

from rpy2.robjects import r

def rvest_json(tiempo_init):
    '''almacena las estadisticas obtenidas de rvest en un fichero json'''
    output = {}

    output['rvest'] = {'RAM % Usage': psutil.virtual_memory()[2], 'Execution time': time.time() - tiempo_init, 'CPU % Usage': psutil.cpu_percent(52.6)}
    with open('./archivos_estadisticas/rvest_stats.json', 'w', encoding='utf8') as file:
        json.dump(output, file, sort_keys=True, ensure_ascii=False, indent=4)

def run_rvest():
    '''ejecucion de rvest, codigo fuente disponible en ./algoritmos_extraccion/run_rvest.R'''
    r('source("./algoritmos_extraccion/run_rvest.R")')

#Probador
tiempo_init = time.time()
output_text = run_rvest()
rvest_json(tiempo_init)